src/prefect/engine.py

```diff
@@ -1420,6 +1425,7 @@ def link_if_trackable(obj: Any) -> None:
         if (type(obj) in UNTRACKABLE_TYPES) or (
             isinstance(obj, int) and (-5 <= obj <= 256)
         ):
+            state.state_details.returns_untrackable_result = True
```

Comment:
We should highlight in this method's docstring that it's mutating the state object. This also suggests that we may be doing things in the incorrect order. Perhaps the state should update this attribute when data is assigned to it. This function can then just check the value of this boolean.

Reply:
Although trackable types are a client-side concept and we don't have a client-side `State` object yet.. so actually probably not worth going down that road yet.